src/Collection/Resources/assets/src/controller.ts

```diff
@@ -0,0 +1,110 @@
+import { Controller } from '@hotwired/stimulus';
+
+const DEFAULT_ITEMS_SELECTOR = ':scope > :is(div, fieldset)';
+
+interface CollectionDataset extends DOMStringMap {
+    prototype: string;
+    currentIndex: string;
+    itemsSelector?: string;
+    addButtonTemplateId?: string;
+    disableAddButton?: string;
+    deleteButtonTemplateId?: string;
+    disableDeleteButton?: string;
+}
+
+enum ButtonType {
+    Add,
+    Delete,
+}
+
+export default class extends Controller {
+    static values = {
+        addButtonTemplateId: '',
+        disableAddButton: false,
+        deleteButtonTemplateId: '',
+        disableDeleteButton: false,
+    };
+
+    connect() {
+        this.connectCollection(this.element as HTMLElement);
+    }
+
+    connectCollection(parent: HTMLElement) {
+        parent.querySelectorAll('[data-prototype]').forEach((el) => {
```

Comment:
I still want to add here a comment that this the most questionable line in my point of view.

In #400 I implemented the JS the way I think it should be implemented that every collection itself has its collection_controller and not that one controller controls all `collections`.

So we even don't need to wrap the whole form with a `data-controller`.

Why I would avoid the `data-prototype` selector here is that for example I have inside my current forms other Form Types using `[data-prototye]` which are even inside collection types, this component currently would identify them also as "collection" which they are not they just use the `data-prototype` for different usecases.

Via a form extension like in #400 we can add a better way by adding `data-controller` to all collection which would solve this problem and even allows to create collection type with a custom stimulus controller like in #90 or #397 also allows extending the current implementaton via events.

Reply:
I'll make this selector configurable and split this controller in two (the main one in case you want all collections to be automatically handled, and a nested one). WDYT?

Reply:
It still is hard to find here a correct selector as the main problems is that the component is registered on the `form` tag and handles multiple instances of collections.

As I did use component libraries over the last years which work very simular to `stimulus` I would really not recommend creating `components` like this which handles `multiple` instances. Instead I would make sure to have a `collection_controller` which handles the a single collection, which should make also the code simpler.

What problem do you see that the `JS` component handles just a `single` collection and adding a `data-controller` via form extension to the collection row like in #400.